### PR TITLE
Port Network Token Function module to the topic/up4-bess rewrite of up4.bess

### PR DESCRIPTION
**Pull request by nevelis**

### CODE REVIEW

**Comment by omecproject** [comment]
Can one of the admins verify this patch?

**Comment by omecproject** [comment]
Can one of the admins verify this patch?

**Comment by omecproject** [comment]
Can one of the admins verify this patch?

File: conf/up4.bess [hunk 1]
```diff
@@ -200,8 +215,15 @@ _in:gate \
 
 # 3. Complete the last part of the DL pipeline
 executeFAR:farTunnelAction \
-    -> etherEncapTrim::GenericDecap(bytes=14) \
-    -> gtpuEncap::GtpuEncap():1 \
+    -> etherEncapTrim::GenericDecap(bytes=14)
+
+# Insert downlink NTF module if enabled
+_in = etherEncapTrim
+if enable_ntf:
```
Comment (ajamshed) [inline conf/up4.bess:222]:
```suggestion
if parser.enable_ntf:
```

File: conf/up4.bess [hunk 2]
```diff
@@ -269,8 +291,16 @@ _in:gate \
 
 # 3. Complete the last part of the UL pipeline
 executeFAR:farForwardAction \
-    -> etherDecapTrim::GenericDecap(bytes=14) -> gtpuDecap::GtpuDecap() \
-    -> sgiEtherAdd::GenericEncap(fields=[
+    -> etherDecapTrim::GenericDecap(bytes=14)
```
Comment (ajamshed) [inline conf/up4.bess:294]:
```suggestion
    -> etherDecapTrim::GenericDecap(bytes=14) \
```

File: conf/up4.bess [hunk 3]
```diff
@@ -269,8 +291,16 @@ _in:gate \
...
+    -> gtpuDecap::GtpuDecap() \
```
Comment (ajamshed) [inline conf/up4.bess:295]:
```suggestion
    -> gtpuDecap::GtpuDecap()
```

File: conf/up4.bess [hunk 4]
```diff
@@ -269,8 +291,16 @@ _in:gate \
...
+
+# Insert uplink NTF module if enabled
+_in = gtpuDecap
+if enable_ntf:
```
Comment (ajamshed) [inline conf/up4.bess:299]:
```suggestion
gate=0
if parser.enable_ntf:
```

File: conf/up4.bess [hunk 5]
```diff
@@ -269,8 +291,16 @@ _in:gate \
...
+    _in:1 -> 0:ntf
```
Comment (ajamshed) [inline conf/up4.bess:300]:
```suggestion
    _in:0 -> 0:ntf
```

File: conf/up4.bess [hunk 6]
```diff
@@ -269,8 +291,16 @@ _in:gate \
...
+    _in = ntf
```
Comment (ajamshed) [inline conf/up4.bess:301]:
```suggestion
    _in = ntf
    gate = 1
```

File: conf/up4.bess [hunk 7]
```diff
@@ -269,8 +291,16 @@ _in:gate \
...
+
+_in:1 -> sgiEtherAdd::GenericEncap(fields=[
```
Comment (ajamshed) [inline conf/up4.bess:303]:
```suggestion
_in:gate -> sgiEtherAdd::GenericEncap(fields=[
```

**Comment by nevelis** [comment]
Will remake against master

**Pull request closed without merging**